k2/csrc/fsa_algo.h

```diff
@@ -158,6 +163,21 @@ Fsa LinearFsa(const Array1<int32_t> &symbols);
  */
 FsaVec LinearFsas(Ragged<int32_t> &symbols);
 
+/* Compute the shortest distance in the tropical semiring, but
+   with the opposite sign. That is, it uses `max` instead of `min.
+
+   @param [in] fsa           The input FSA. It is on CPU and must be
+                             topologically sorted.
+   @param [out] arc_map      The arc indexes of the best path from the start
+                             state to the final state. It is empty if there
+                             is no such path.
+   @return The shortest distance from the start state to the final state.
+           Returns negative infinity if there is no such path.
+ */
+double ShortestDistance(Fsa &fsa, Array1<int32_t> *arc_map = nullptr);
```

Comment:
Better to add documentation to say that we only get one of those best paths (if there are multiple best paths)